### PR TITLE
fix: broken login when user has org units associated

### PR DESCRIPTION
**Pull request by netroms**
### Summary
Fixes issue during user details creation, when hydrating Hibernate managed OrgUnits to populate the UserDetails object.
Since there are many calls to `CurrentUserUtil.getCurrentUserDetails()` in the getters themselves in several places, this will cause an exception since we are not logged in yet. Since we only need the OrgUnit IDs during creation of the UserDetails, separate data store/SQL is made to fetch only the UIDs.

Also makes some startup data populators use 'current user safe' service access methods, i.e. running as SystemUser.

## Automatic test
OrganisationUnitStoreTest#testOrganisationUnitsUidByUser()

### CODE REVIEW

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserDetails.java [hunk 1]
```diff
@@ -171,8 +188,14 @@ default boolean isInUserDataHierarchy(String orgUnitPath) {
...
-    if (orgUnitPath == null) return false;
-    for (String uid : orgUnitPath.split("/")) if (orgUnitIds.contains(uid)) return true;
+    if (orgUnitPath == null) {
+      return false;
+    }
+    for (String uid : orgUnitPath.split("/")) {
+      if (orgUnitIds.contains(uid)) {
+        return true;
+      }
+    }
```
Comment (jbee) [inline dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserDetails.java:198]: Do you have your IDE configured to insert brackets for all block statements?

**Review by jbee — COMMENTED** [review]
Do I understand correctly that the issue is that a `User` used to create a `UserDetails` might be detached and therefore not correctly populated when it comes to these lists so therefore you changed this to query it directly?

File: dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserDetails.java (hunk 1 above)
Comment (netroms) [reply dhis-2/dhis-api/src/main/java/org/hisp/dhis/user/UserDetails.java:198]: Sorry about that, it might be my IDE yeah. I will revert it.

**Review by vietnguyen — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/organisationunit/hibernate/HibernateOrganisationUnitStore.java [hunk 2]
```diff
@@ -86,6 +86,35 @@ public HibernateOrganisationUnitStore(
...
+
+  private static String getOrgUnitTablesUids(String username, String orgUnitTableName) {
+    return """
+        SELECT ou.uid
+        FROM organisationunit ou
+                 JOIN %s um ON ou.organisationunitid = um.organisationunitid
+                 JOIN userinfo ui ON um.userinfoid = ui.userinfoid
+        WHERE ui.username = '%s';
+        """
+        .formatted(orgUnitTableName, username);
```
Comment (vietnguyen) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/organisationunit/hibernate/HibernateOrganisationUnitStore.java:115]: maybe we could have one single query for those properties.

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java [hunk 3]
```diff
@@ -909,8 +910,21 @@ public UserDetails createUserDetails(User user) {
 
     Map<String, Serializable> userSettings = userSettingService.getUserSettingsAsMap(user);
 
+    List<String> organisationUnitsUidsByUser =
+        organisationUnitService.getOrganisationUnitsUidsByUser(user.getUsername());
+    List<String> searchOrganisationUnitsUidsByUser =
```
Comment (vietnguyen) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/user/DefaultUserService.java:915]: the user.organisationUnit settings should not be updated regularly, maybe we could use a cache here.

**Pull request merged**